src/librustdoc/test.rs

```diff
@@ -262,17 +262,19 @@ fn runtest(test: &str, cratename: &str, cfgs: Vec<String>, libs: SearchPaths,
         control.after_analysis.stop = Compilation::Stop;
     }
 
-    let res = panic::catch_unwind(AssertUnwindSafe(|| {
```

Comment:
I guess it's a question of point of view? For me it's better this way but if someone else agrees with you, then I'll do it again in two steps.